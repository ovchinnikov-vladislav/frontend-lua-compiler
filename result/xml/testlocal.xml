<BlockNode>
  <StatList>
    <FunctionDefinition>
      <FunctionNameVariable>
        <Name name="a"/>
      </FunctionNameVariable>
      <FunctionBody>
        <ParList varPar="true">
          <NameList>
            <Name name="a"/>
          </NameList>
        </ParList>
        <BlockNode>
          <StatList/>
        </BlockNode>
      </FunctionBody>
    </FunctionDefinition>
  </StatList>
</BlockNode>
