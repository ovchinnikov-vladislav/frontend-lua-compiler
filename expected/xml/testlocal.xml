<BlockNode>
  <StatList>
    <IfThenElse>
      <BinaryOperation operation="EQUAL">
        <Variable var="a"/>
        <NumeralExpression number="5" type="INTEGER"/>
      </BinaryOperation>
      <BlockNode>
        <StatList>
          <Assignment>
            <VarList>
              <Variable var="a"/>
            </VarList>
            <ExpList>
              <BinaryOperation operation="MUL">
                <NumeralExpression number="3" type="INTEGER"/>
                <NumeralExpression number="5.0" type="DOUBLE"/>
              </BinaryOperation>
            </ExpList>
          </Assignment>
          <Assignment>
            <VarList>
              <Variable var="b"/>
            </VarList>
            <ExpList>
              <BinaryOperation operation="MUL">
                <NumeralExpression number="3" type="INTEGER"/>
                <NumeralExpression number="7" type="INTEGER"/>
              </BinaryOperation>
            </ExpList>
          </Assignment>
          <Assignment>
            <VarList>
              <Variable var="f"/>
            </VarList>
            <ExpList>
              <FunctionExpression varArgs="false">
                <NameList/>
                <Block>
                  <StatList>
                    <Assignment>
                      <VarList>
                        <Variable var="a"/>
                      </VarList>
                      <ExpList>
                        <BinaryOperation operation="ADD">
                          <Variable var="a"/>
                          <BinaryOperation operation="MUL">
                            <Variable var="b"/>
                            <Variable var="a"/>
                          </BinaryOperation>
                        </BinaryOperation>
                      </ExpList>
                    </Assignment>
                  </StatList>
                  <Return>
                    <ExpList>
                      <Variable var="a"/>
                    </ExpList>
                  </Return>
                </Block>
              </FunctionExpression>
            </ExpList>
          </Assignment>
        </StatList>
      </BlockNode>
      <BlockNode/>
    </IfThenElse>
    <Assignment>
      <VarList>
        <Variable var="c"/>
      </VarList>
      <ExpList>
        <BinaryOperation operation="OR">
          <BinaryOperation operation="MUL">
            <NumeralExpression number="5" type="INTEGER"/>
            <NumeralExpression number="5.0" type="DOUBLE"/>
          </BinaryOperation>
          <BinaryOperation operation="MUL">
            <NumeralExpression number="3" type="INTEGER"/>
            <NumeralExpression number="10" type="INTEGER"/>
          </BinaryOperation>
        </BinaryOperation>
      </ExpList>
    </Assignment>
    <Assignment>
      <VarList>
        <Variable var="b"/>
      </VarList>
      <ExpList>
        <Variable var="c"/>
      </ExpList>
    </Assignment>
  </StatList>
</BlockNode>
